File: Python/k03.py
from random import Random, randint

class k03(object):
    
    def execute(self, num):
        getattr(self, 'question%d' % num)()

    def question1(self):
        try:
            print('年齢を入力してください。＞')
            age = int(input())
            if age < 20:
                print('未成年なので購入できません。')
        except ValueError as ve:
            print(ve)

    def question2(self):
        try:
            print('身長を入力してください。＞')
            height = float(input())
            height /= 100.0
            print('体重を入力してください。＞')
            weight = float(input())
            standard = height * height * 22.0
            print('あなたの標準体重は{}です。'.format(standard))

            if weight > standard and (weight - standard) / standard * 100 > 14:
                print('太り気味です。')
            elif weight < standard and (weight - standard) / standard * 100 < -14:
                print('痩せ気味です。')
            else:
                print('普通ですね。')

        except ValueError as ve:
            print(ve)

    def question3(self):
        try:
            generator = Random()
            number = generator.randint(0, 99)
            print('０から９９の範囲の数値が決定されました。')
            print('決められた数値を予想し、この数値よりも大きな値を入力してください＞')
            
            guess = int(input())
            print('決められた数値は{}です。'.format(number))
            print('正解です。' if guess > number else '不正解です。')
        except ValueError as ve:
            print(ve)

    def question4(self):
        try:
            generator = Random()
            number = generator.randint(0, 99)
            print('０から９９の範囲の数値が決定されました。')
            print('決められた数値を予想し、この数値よりも大きな値を入力してください＞')
            
            guess = int(input())
            print('決められた数値は{}です。'.format(number))

            if guess > 100 or guess < 0:
                print('反則です！')
            elif guess > number and guess - number <= 10:
                print('大正解！')
            elif guess < number and guess - number >= -10:
                print('惜しい！')
            elif guess == number:
                print('お見事！')
            else:
                print('正解です。' if guess > number else '不正解です。')
        except ValueError as ve:
            print(ve)

File: Go/K03.go
package main

import (
	"fmt"
	"math/rand"
	"strconv"
)

type K03 struct {

}

func (k03 K03) question1() {
	fmt.Print("年齢を入力してください。＞")
	var input string
	_, _ = fmt.Scanln(&input)
	age, _ := strconv.Atoi(input)

	if age < 20 {
		fmt.Println("未成年なので購入できません。")
	}
}

func (k03 K03) question2() {
	fmt.Print("身長を入力してください。＞")
	var input string
	_, _ = fmt.Scanln(&input)
	_result, _ := strconv.Atoi(input)
	height := float32(_result)
	height /= 100.0

	fmt.Print("体重を入力してください。＞")
	_, _ = fmt.Scanln(&input)
	_result, _ = strconv.Atoi(input)
	weight := float32(_result)

	standard := height * height * 22.0
	fmt.Println(fmt.Sprintf("あなたの標準体重は%fです。", standard))

	if weight > standard && (weight - standard) / standard * 100.0 > 14.0 {
		fmt.Println("太り気味です。")
	} else if weight < standard && (weight - standard) / standard * 100.0 < -14.0 {
		fmt.Println("痩せ気味です。")
	} else {
		fmt.Println("普通ですね。")
	}
}

func (k03 K03) question3() {
	randomNumber := rand.Intn(100)
	fmt.Println("０から９９の範囲の数値が決定されました。")
	fmt.Print("決められた数値を予想し、この数値よりも大きな値を入力してください＞")

	var input string
	_, _ = fmt.Scanln(&input)
	guess, _ := strconv.Atoi(input)

	fmt.Println(fmt.Sprintf("決められた数値は%dです", randomNumber))
	if guess > randomNumber {
		fmt.Println("正解です。")
	} else {
		fmt.Println("不正解です。")
	}
}

func (k03 K03) question4() {
	randomNumber := rand.Intn(100)
	fmt.Println("０から９９の範囲の数値が決定されました。")
	fmt.Print("決められた数値を予想し、この数値よりも大きな値を入力してください＞")

	var input string
	_, _ = fmt.Scanln(&input)
	guess, _ := strconv.Atoi(input)

	fmt.Println(fmt.Sprintf("決められた数値は%dです", randomNumber))

	if guess > 100 || guess < 0 {
		fmt.Println("反則です！")
	} else if guess > randomNumber && guess - randomNumber <= 10 {
		fmt.Println("大正解です！")
	} else if guess < randomNumber && guess - randomNumber >= -10 {
		fmt.Println("惜しい！")
	} else if guess == randomNumber {
		fmt.Println("お見事！")
	} else {
		if guess > randomNumber {
			fmt.Println("正解です。")
		} else {
			fmt.Println("不正解です。")
		}
	}
}

func (k03 K03) execute(num int) {
	switch num {
	case 1:
		k03.question1()
	case 2:
		k03.question2()
	case 3:
		k03.question3()
	case 4:
		k03.question4()
	}
}